compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -3354,7 +3354,8 @@ private void validateActionInvocation(Location pos, BLangInvocation iExpr) {
         if (iExpr.expr != null) {
             final NodeKind clientNodeKind = iExpr.expr.getKind();
             // Validation against node kind.
-            if (clientNodeKind != NodeKind.SIMPLE_VARIABLE_REF && clientNodeKind != NodeKind.FIELD_BASED_ACCESS_EXPR) {
+            if (clientNodeKind != NodeKind.SIMPLE_VARIABLE_REF && clientNodeKind != NodeKind.FIELD_BASED_ACCESS_EXPR &&
+            clientNodeKind != NodeKind.GROUP_EXPR) {
```

Comment:
With the above change,
following compiles fine.
```ballerina
type A record {
    function() returns int f = function() returns int => 1;
};

public function main() {
    var f = start (foo()).f();
}

function foo() returns A {
    return {};
}
```
but the following results in `action invocation as an expression not allowed here`
```ballerina
type A record {
    function() returns int f = function() returns int => 1;
};

public function main() {
    var f = start foo().f();
}

function foo() returns A {
    return {};
}
```

Reply:
Will add support for that. And I think with the new changes this error messages no longer valid since now action invocation as an expression allowed. Shall I change the error message?

Reply:
@rpjayasekara shall we raise this in the dev-compiler channel in slack, according to the spec I don't see why we have constrained this to `start a.b()`,  and `start self.a.b()` only.

note: Below breaks are in beta-2, not in this PR.
```ballerina
class className {
    C c = new;
    function bar() {

    }

    function foo() {
        var x = start self.bar();
        var y = start self.c.bar(); // special cased to allow this, but not below 2
        var z = start self.c.c.bar(); // action invocation as an expression not allowed here(BCE3038)
        var z1 = start self.c.c.c.bar(); // action invocation as an expression not allowed here(BCE3038)
        
        var n = new className();
        // var p = start (n).bar(); // breaks the compiler
    }
}

class C {
    className c = new;
    function bar() {

    }
}

function bar() {

}

function f() returns (object { function foo() returns int; }) {
    return object {function foo() returns int => 1;};
}

public function main() returns error? {
    className o = new className();
    var a = start bar();
    var b = start o.bar();
    // var c = start (o).bar(); // breaks the compiler
    // var d = start f().foo(); // brewaks the compiler
}
```

Reply:
Will raise the issue in slack